bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpClientFactory.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import org.eclipse.jetty.client.HttpClient;
+
+/**
+ * Factory class to create jetty http clients
+ *
+ * @author Michael Bock - initial API
+ */
+public interface HttpClientFactory {
+
+    /**
+     * Creates a new jetty http client..
+     * The returned client is already started but must be stopped
+     * by the consumer after usage. The client lifecycle should be the same as for the consumer.
+     * DO NOT CREATE NEW CLIENTS FOR EACH REQUEST!
+     * 
+     * @param consumerName the for identifying the consumer in the jetty thread pool.
+     *            Must be between 4 and 20 characters long and must contain only the following characters [a-zA-Z0-9-_]
+     * @param endpoint the desired endpoint, protocol and host are sufficient
+     * @return the jetty client
+     * @throws NullPointerException if {@code endpoint} or {@code consumerName} is {@code null}
```

Comment:
Instead of this, you should add a `@NonNullByDefault` annotation to the interface (see also [A.12](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#a-code-style)).

Reply:
Müsste das nicht @NonNull sein?

Reply:
No, it should be `@NonNullByDefault` annotation on the interface, not on the method params.

Reply:
This line btw seems outdated and should imho be removed